src/include/interval_set.h

```diff
@@ -406,35 +409,35 @@ class interval_set {
   bool empty() const {
     return m.empty();
   }
-  T range_start() const {
+  const_iterator range_start() const {
```

Comment:
i don't understand this change. `p->first` should be of `T`, right?

Reply:
Yes, oops.